pkg/kubelet/token/token_manager.go

```diff
@@ -146,6 +149,14 @@ func (m *Manager) requiresRefresh(tr *authenticationv1.TokenRequest) bool {
 }
 
 // keys should be nonconfidential and safe to log
-func keyFunc(name, namespace string, tr *authenticationv1.TokenRequest) string {
-	return fmt.Sprintf("%q/%q/%#v", name, namespace, tr.Spec)
+func keyFunc(name, namespace string, tr *authenticationv1.TokenRequest) (string, error) {
+	if tr.Spec.ExpirationSeconds == nil {
+		return "", fmt.Errorf("expiration seconds was nil for tr: %#v", tr)
+	}
+	if tr.Spec.BoundObjectRef == nil {
+		return "", fmt.Errorf("bound object was nil for tr: %#v", tr)
+	}
+
+	specKey := fmt.Sprintf("%#v:%d:%#v", tr.Spec.Audiences, *tr.Spec.ExpirationSeconds, *tr.Spec.BoundObjectRef)
```

Comment:
Call fmt.Sprintf only once, with all of the vars.
Also, any reason to use `:` here as separator vs `/` below?